compiler/luci-interpreter/src/kernels/Logistic.test.cpp

```diff
@@ -25,33 +25,63 @@ namespace
 {
 
 using namespace testing;
+const float kQuantizedTolerance = 2 * (1. / 256);
 
-TEST(LogisticTest, Float)
+template <typename T>
+void Check(std::initializer_list<int32_t> input_shape, std::initializer_list<int32_t> output_shape,
+           std::initializer_list<float> input_data, std::initializer_list<float> output_data,
+           float input_min, float input_max, float output_min, float output_max,
+           DataType element_type)
 {
-  Shape input_shape{1, 2, 4, 1};
-  std::vector<float> input_data{
-      0, -6, 2,  4, //
-      3, -2, 10, 1, //
-  };
-  Tensor input_tensor = makeInputTensor<DataType::FLOAT32>(input_shape, input_data);
-  Tensor output_tensor = makeOutputTensor(DataType::FLOAT32);
+  std::pair<float, int32_t> input_quant_params = quantizationParams<T>(input_min, input_max);
+  std::pair<float, int32_t> output_quant_params = quantizationParams<T>(output_min, output_max);
+  std::vector<T> quantized_input_value =
+      quantize<T>(input_data, input_quant_params.first, input_quant_params.second);
+  Tensor input_tensor{
+      element_type, input_shape, {{input_quant_params.first}, {input_quant_params.second}}, ""};
+  input_tensor.writeData(quantized_input_value.data(), quantized_input_value.size() * sizeof(T));
+  Tensor output_tensor =
+      makeOutputTensor(element_type, output_quant_params.first, output_quant_params.second);
 
   Logistic kernel(&input_tensor, &output_tensor);
   kernel.configure();
   kernel.execute();
 
-  std::vector<float> ref_output_data{
-      0.5,      0.002473, 0.880797, 0.982014, //
-      0.952574, 0.119203, 0.999955, 0.731059, //
-  };
-  EXPECT_THAT(extractTensorData<float>(output_tensor),
-              ElementsAreArray(ArrayFloatNear(ref_output_data)));
-  // TODO make a Shape checking of output_tensor.
+  EXPECT_THAT(dequantize(extractTensorData<T>(output_tensor), output_tensor.scale(),
```

Comment:
Is it possible to separate this check code according to the `element_type`? E.g., not dequantizing values for float.

Reply:
Okay, I fix like what you suggested.